modules/NuGetPackageVerifier/console/Rules/PackageVersionMatchesAssemblyVersionRule.cs

```diff
@@ -30,10 +30,8 @@ public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context
                        infoVersion,
                        context.Metadata.Version,
                        context.Metadata.Id);
-                    continue;
                 }
-
-                if (!VersionEquals(context.Metadata.Version, assemblyInformationalNuGetVersion))
+                else if (!VersionEquals(context.Metadata.Version, assemblyInformationalNuGetVersion))
```

Comment:
This should still stay `if`? (like if both this and the above condition fail). Would be good add a test for this (if there are tests already for these).

Reply:
Hmm, I was thinking "Oh it will just give two of the same error message", but if there's two different ways it will fail I think I agree with @kichalla that it should fail all of them up-front.

Reply:
Updated